petastorm/reader.py

```diff
@@ -421,6 +421,8 @@ def __init__(self, pyarrow_filesystem, dataset_path, schema_fields=None,
             fields = schema_fields if isinstance(schema_fields, collections.Iterable) else None
 
         storage_schema = stored_schema.create_schema_view(fields) if fields else stored_schema
+        if len(storage_schema.fields) == 0:
```

Comment:
Nit: `if not storage_schema.fields` should work here since the `.fields` attribute is an `OrderedDict`.

Reply:
That's true. I don't mind either, but frankly prefer the explicit check since otherwise it may be interpreted as checking if the attribute is None.